File: example/code_ocaml/src/main/ocaml/app/Logger.ml
let log_current_test received n =
    Printf.printf "> fibonacci(%i) = %i\n" n received;;

let log_no_match expected =
    Printf.eprintf "----------------------------------------------------\n";
    Printf.eprintf "The result does not match the expected value: %i\n" expected;
    Printf.eprintf "----------------------------------------------------\n";;

let log_exception e =
    Printf.eprintf "----------------------------------------------------\n";
    Printf.eprintf "An error occurred during runtime.\n";
    Printf.eprintf "Details:\n";
    let msg = Printexc.to_string e in
    Printf.eprintf "%s\n" msg;;

let log_success () =
    Printf.printf "Everything went well, good job!\n";
    Printf.printf "----------------------------------------------------\n";;

File: example/code_java_maven/src/main/java/app/Logger.java
package app;

public class Logger {

    public static void logCurrentTest(String operation, int expected, int received, int a, int b) {
        System.err.println("> " + operation + "(" + a + "," + b + ") = " + received);
        System.err.println("> expected = " + expected);
        System.err.println("----------------------------------------------------");
    }

    public static void logSuccess() {
        System.out.println("Everything went well, good job!");
        System.out.println("----------------------------------------------------");
    }

    public static void logException(Throwable throwable) {
        System.err.println("----------------------------------------------------");
        System.err.println("Something bad happened!");
        System.err.println(throwable.getMessage());
        throwable.printStackTrace();
        System.err.println("----------------------------------------------------");
    }
}
